src/resolvers/taxonomyResolvers.ts

```diff
@@ -0,0 +1,180 @@
+/**
+ * Copyright (c) 2024-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import { IArticleV2 } from "@ndla/types-backend/article-api";
+import { ISubjectPageData } from "@ndla/types-backend/frontpage-api";
+import { Node, TaxonomyContext } from "@ndla/types-taxonomy";
+import {
+  fetchChildren,
+  fetchLK20CompetenceGoalSet,
+  fetchLearningpath,
+  fetchNode,
+  fetchNodeByContentUri,
+  queryNodes,
+} from "../api";
+import {
+  GQLLearningpath,
+  GQLMeta,
+  GQLNode,
+  GQLNodeChildrenArgs,
+  GQLQueryNodeArgs,
+  GQLQueryNodeByArticleIdArgs,
+  GQLQueryNodesArgs,
+  GQLTaxonomyEntity,
+} from "../types/schema";
+import { nodeToTaxonomyEntity } from "../utils/apiHelpers";
+import {
+  articleToMeta,
+  filterMissingArticles,
+  getArticleIdFromUrn,
+  getLearningpathIdFromUrn,
+} from "../utils/articleHelpers";
+
+const pickContextValues = (node: Node, language: string, rootId?: string, parentId?: string) => {
+  const visibleContexts = node.contexts.filter((c) => c.isVisible && !c.rootId.startsWith("urn:programme"));
+  // A resource can have several context in one subject. rootId and parentId helps picking the right one.
+  const rootContexts = rootId ? visibleContexts.filter((c) => c.rootId === rootId) : visibleContexts;
+  const parentContexts = parentId ? rootContexts.filter((c) => c.parentIds.includes(parentId)) : rootContexts;
+
+  const selectedCtx = parentContexts?.[0];
+  const path = selectedCtx?.path || node.path;
+  const rank = selectedCtx?.rank;
+  const contextId = selectedCtx?.contextId;
+  const relevanceId = selectedCtx?.relevanceId;
+  const entity = nodeToTaxonomyEntity({ ...node, contexts: visibleContexts }, language);
+  return { ...entity, contextId, path, rank, relevanceId };
+};
```

Comment:
Går det ikke an å regne ut dette i tax?

Reply:
Det brukes ved henting av node fra artikkelid, men det er ikkje sikkert vi trenger det for det er jo iframe-visning.

Reply:
Samtidig kan du risikere å få en kontekst som er skjult dersom du henter en ressurs som er plassert under et emne som er skjult. Då vil ikkje urlen som vises i iframe-visninga være en gyldig url.